@Inject BeanManager in @QuarkusTest annotated class does not work

Hi @mkouba,

This is in regards to the issue we discussed on Zulip.

I cannot `@Inject` `BeanManager` into a `@QuarkusTest` annotated class. When I have a test class like this one:

```java
@QuarkusTest
public class EventsBypassTest {

    @Inject
    BeanManager bm;

    // ...
}
```

It fails with the following error:

```plain
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.007 s <<< FAILURE! - in com.github.sarxos.abberwoult.EventsBypassTest
[ERROR] test_startupEventReceived  Time elapsed: 0.007 s  <<< ERROR!
org.junit.jupiter.api.extension.TestInstantiationException: TestInstanceFactory [io.quarkus.test.junit.QuarkusTestExtension] failed to instantiate test class [com.github.sarxos.abberwoult.EventsBypassTest]: Failed to inject field javax.enterprise.inject.spi.BeanManager com.github.sarxos.abberwoult.EventsBypassTest.bm
Caused by: java.lang.RuntimeException: Failed to inject field javax.enterprise.inject.spi.BeanManager com.github.sarxos.abberwoult.EventsBypassTest.bm
Caused by: javax.enterprise.inject.UnsatisfiedResolutionException
```

In the same time I can get BeanManager from CDI (and this is what I did as a workaround to the problem):

```java
@Test
public void test_eventsReceived() throws InterruptedException {
    final BeanManager bm = CDI.current().getBeanManager();
    // ... use bean manager here
}
```

This happens on Quarkus `0.14.0`.

--- a/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcTestResourceProvider.java
+++ b/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcTestResourceProvider.java
@@ -24,19 +24,26 @@ public class ArcTestResourceProvider implements TestResourceProvider {
         while (c != Object.class) {
             for (Field field : c.getDeclaredFields()) {
                 if (field.isAnnotationPresent(Inject.class) || field.isAnnotationPresent(ConfigProperty.class)) {
+                    Object instance;
+
                     try {
-                        Set<Annotation> qualifiers = new HashSet<>();
-                        Set<Annotation> annotations = new HashSet<>();
-                        for (Annotation a : field.getAnnotations()) {
-                            annotations.add(a);
-                            if (a.annotationType().isAnnotationPresent(Qualifier.class)) {
-                                qualifiers.add(a);
+                        if (field.getType().equals(BeanManager.class)) {
+                            instance = beanManager;
+                        } else {
+                            Set<Annotation> qualifiers = new HashSet<>();
+                            Set<Annotation> annotations = new HashSet<>();
+                            for (Annotation a : field.getAnnotations()) {
+                                annotations.add(a);
+                                if (a.annotationType().isAnnotationPresent(Qualifier.class)) {
+                                    qualifiers.add(a);
+                                }
                             }
+                            instance = beanManager.getInjectableReference(
+                                    new InjectionPointImpl(field.getGenericType(), field.getGenericType(),
+                                            qualifiers, null, annotations, field, -1),
+                                    beanManager.createCreationalContext(null));
                         }
-                        Object instance = beanManager.getInjectableReference(
-                                new InjectionPointImpl(field.getGenericType(), field.getGenericType(),
-                                        qualifiers, null, annotations, field, -1),
-                                beanManager.createCreationalContext(null));
+
                         // Set the field value
                         field.setAccessible(true);
                         try {

--- a/integration-tests/main/src/test/java/io/quarkus/it/main/BeanManagerInjectionInTestCase.java
+++ b/integration-tests/main/src/test/java/io/quarkus/it/main/BeanManagerInjectionInTestCase.java
@@ -1,0 +1,23 @@
+package io.quarkus.it.main;
+
+import javax.enterprise.inject.spi.BeanManager;
+import javax.inject.Inject;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.it.rest.ExternalService;
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public class BeanManagerInjectionInTestCase {
+
+    @Inject
+    BeanManager beanManager;
+
+    @Test
+    public void testInjection() {
+        Assertions.assertNotNull(beanManager);
+        Assertions.assertFalse(beanManager.getBeans(ExternalService.class).isEmpty());
+    }
+}

